Update readme (#58)
embroider-build/release-plan: README.md

> This package was originally developed to help release [Embroider](https://github.com/embroider-build/embroider) and was extracted so everyone can use it 🎉

Reviewer: This probably isn't necessarry as part of the intro readme, especially if it's ready for other projects to use.

Author: true -- I'll move to the bottom

## Features

> - no manual steps[^after-initial-setup]

Reviewer: ```suggestion
- ✨ No manual steps[^after-initial-setup]
```

  - release from _anywhere_[^release-anywhere]

> - everything is automated via GH Action Workflows[^gh-by-default]

Reviewer: ```suggestion
- 🤖 Everything is automated via GH Action Workflows[^gh-by-default]
```

> - no need to give maintainers access to npm[^npm-config]

Reviewer: ```suggestion
- 🔐 No need to give maintainers access to npm[^npm-config]
```

> - pnpm and monorepo support _by default_

Reviewer: ```suggestion
- 📦 PNPM and monorepo support _by default_
```

  - also supports non-pnpm repos

> - release preview PR 

Reviewer: ```suggestion
- 🚀 Release preview PR 
```

>   - release via merge of the releae preview PR

Reviewer: ```suggestion
  - release via merge of the release preview PR
```

> - automatic changelog

Reviewer: ```suggestion
- 📝 Automatic changelog
```

  - changelog entries editable by editing PR titles

This allows folks with GH Maintainer access to preview and release to npm without actually needing to have any keys locally on their machine (or access to the npm package, which makes management of the repo separate from the npm package easy and encouraged).

[^gh-by-default]: While GitHub Actions is the default setup via [`create-release-plan-setup`][gh-create]
[^after-initial-setup]: after initial setup, which is already mostly automated.
[^npm-config]: The NPM token can be granular, and the package needs to have its "Publishing access" configured to either "Don't require two-factor authentication" or, "Require two-factor authentication or an automation or granular access token" (recommended). This can be configured under `https://www.npmjs.com/package/{package-name}/access`

> [^release-anywhere]: You can release from bed, the mall, the percelain, anywhere! No need for a computer! (assuming you have a pocket computer (phone))

Reviewer: ```suggestion
[^release-anywhere]: You can release from bed, the mall, the porcelain, anywhere! No need for a computer! (assuming you have a smart phone)
```

Author: what's wrong with pocket computer? :sweat_smile:

Reviewer: lol I don't like brackets inside brackes 😄

Author: I think we should all start calling them pocket computers, as none of ever actually use the phone part :sweat_smile:

> 5. Once the PR is merged, in a clean local repo at the merge commit, run `npx release-plan publish`. if you need an `otp` for your release you can provide that to the `publish` command like this `npx release-plan publish --otp=123456`

Reviewer: ```suggestion
5. Once the PR is merged, in a clean local repo at the merge commit, run `npx release-plan publish`. If you need an `otp` for your release you can provide that to the `publish` command like this `npx release-plan publish --otp=123456`
```

    

## Comparison

> Note that we take the stance of reducing friction for new contributors (regardless of how new to GitHub they are), we want to optimize for the contribution, and assume that mantainers can handle a little bit of process -- and with this stance, changesets is the _most work_ out of this comparison.

Reviewer: ```suggestion
Note that we take the stance of reducing friction for new contributors (regardless of how new to GitHub they are), we want to optimize for the contribution, and assume that maintainers can handle a little bit of process -- and with this stance, changesets is the _most work_ out of this comparison.
```
